lib/data_hygiene/delete_topics_without_subscribers.rb

```diff
@@ -0,0 +1,116 @@
+require 'thread'
+
+module DataHygiene
+  class DeleteTopicsWithoutSubscribers
+    attr_reader :client
+    delegate :puts, to: :@output
+    delegate :gets, to: :@input
+
+    def initialize(client: Services.gov_delivery, input: STDIN, output: STDOUT)
+      @client = client
+      @input = input
+      @output = output
+    end
+
+    def call
+      puts "#{with_zero_subscribers.count} subscriber lists without subscribers"
+      with_zero_subscribers.each { |sl| puts "#{sl.gov_delivery_id} - #{sl.title}"}
+      if with_gov_delivery_error.count > 0
+        puts ''
+        puts "#{with_gov_delivery_error.count} subscriber lists with errors - THESE ARE NOT BEING DELETED"
+        with_gov_delivery_error.each { |sl| puts "#{sl.gov_delivery_id} - #{sl.title}"}
+      end
+
+      puts ''
+      puts "#{with_zero_subscribers.count} subscriber lists without subscribers, enter `delete` to delete from database and GovDelivery: "
+      command = gets
+
+      if command.chomp == 'delete'
+        with_zero_subscribers.each do |subscriber_list|
+          delete_topic(subscriber_list)
+        end
+      end
+    end
+
+    def subscriber_lists
+      @subscriber_lists ||= ThreadedSubscriberCountRetriever.new(client: client, output: @output).call
+    end
+
+    def with_zero_subscribers
+      @with_zero_subscribers ||= subscriber_lists
+        .select { |c, s| c&.zero? }
+        .map(&:last)
+    end
+
+    def with_gov_delivery_error
+      @with_gov_delivery_error ||= subscriber_lists
+        .select { |c, s| c.nil? }
+        .map(&:last)
+    end
+
+  private
+
+    def delete_topic(subscriber_list)
+      topic = client.fetch_topic(subscriber_list.gov_delivery_id)
+
+      if topic && topic['subscribers_count'] == 0
+        puts "Deleting: #{subscriber_list.gov_delivery_id}"
+        delete_topic_with_error_catching(subscriber_list.gov_delivery_id)
```

Comment:
We're actually only deleting from GovDelivery here, and not from the local database.